ast/expressions.go

```diff
@@ -606,7 +606,29 @@ type PatternInExpr struct {
 
 // Restore implements Node interface.
 func (n *PatternInExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if err := n.Expr.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while restore PatternInExpr.Expr")
+	}
+	if n.Not {
+		ctx.WriteKeyWord(" NOT IN (")
+	} else {
+		ctx.WriteKeyWord(" IN (")
+	}
+	for i, expr := range n.List {
+		if err := expr.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while restore PatternInExpr.List")
+		}
+		if i != len(n.List)-1 {
```

Comment:
```
if i !=0 {
    ctx.WritePlain(",")
}
```
what about add this before L618 and remove `if i != len(n.List)-1 {..}`

Reply:
btw, I changed the similar code in `PatternInExpr.Format`